ircDDBGateway/APRSTransmit/APRSTransmitAppD.cpp

```diff
@@ -79,8 +79,15 @@ int main(int argc, char** argv)
 	}
 
 	wxString repeater = parser.GetParam(0U);
+	if (repeater.length() != LONG_CALLSIGN_LENGTH) {
+		wxString callErrorMsg;
+		callErrorMsg << wxT("Invalid repeater call. ") << repeater << wxT("Call must be ") << LONG_CALLSIGN_LENGTH << wxT(" characters long.\n");
```

Comment:
> On May 10, 2016, at 6:23 AM, Geoffrey Merck <notifications@github.com> wrote:
> 
> In ircDDBGateway/APRSTransmit/APRSTransmitAppD.cpp:
> 
> > @@ -79,8 +79,15 @@ int main(int argc, char** argv)
> >  	}
> >  
> >  	wxString repeater = parser.GetParam(0U);
> > +	if (repeater.length() != LONG_CALLSIGN_LENGTH) {
> > +		wxString callErrorMsg;
> > +		callErrorMsg << wxT("Invalid repeater call. ") << repeater << wxT("Call must be ") << LONG_CALLSIGN_LENGTH << wxT(" characters long.\n");
> 
> I have the same understanding. However I think using wxS shall be the way to go as it saves some CPU since conversion is made at compile time rather than runtime.

Like I said, I think that’s a premature optimization.  It’s much easier to just use straight strings.  That section of code *only* going to get used when there’s an invalid callsign into the application.  Additionally, the function immediately exits from the program.  It’s certainly not a critical path that needs to be optimized.

--
Jeremy McDermond (NH6Z)
Xenotropic Systems
mcdermj@xenotropic.com